QueryBuilder/AnonymizedQueryBuilder.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/data-protection-bundle project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\DataProtectionBundle\QueryBuilder;
+
+use Ekino\DataProtectionBundle\Annotations\AnonymizedProperty;
+use Ekino\DataProtectionBundle\Meta\AnonymizedMetadata;
+
+/**
+ * Class AnonymizedQueryBuilder.
+ * This class aims to build an anonymizing query based on an AnonymizedMetadata object.
+ *
+ * @author Benoit Mazière <benoit.maziere@ekino.com>
+ */
+final class AnonymizedQueryBuilder
+{
+    /**
+     * return string
+     */
+    public function buildQuery(AnonymizedMetadata $anonymizedMetadata): string
+    {
+        $query = '';
+
+        if ($anonymizedMetadata->getAnonymizedEntity()->isTruncateAction()) {
+            $query = $this->buildTruncateQuery($anonymizedMetadata);
+        } elseif ($anonymizedMetadata->getAnonymizedEntity()->isAnonymizeAction()) {
+            $query = $this->buildAnonymizeQuery($anonymizedMetadata);
+        }
+
+        $exceptWhereClause = $anonymizedMetadata->getAnonymizedEntity()->getExceptWhereClause() ?? '';
+
+        return $exceptWhereClause ? sprintf('%s WHERE %s', $query, $exceptWhereClause) : $query;
+    }
+
+    private function buildTruncateQuery(AnonymizedMetadata $anonymizedMetadata): string
+    {
+        return sprintf('DELETE * FROM %s', $anonymizedMetadata->getClassMetadata()->getTableName());
```

Comment:
truncate or delete? :)

Reply:
that's kind of truncate except in case you have a where clause. Maybe we should use truncate if where clause is null and delete if not. WDYT?